nv_peer_mem.c

```diff
@@ -251,6 +268,7 @@ static int nv_dma_map(struct sg_table *sg_head, void *context,
 			peer_err("error, incompatible dma mapping version 0x%08x\n",
 				 dma_mapping->version);
 			nvidia_p2p_dma_unmap_pages(pdev, page_table, dma_mapping);
+			nvidia_p2p_free_dma_mapping(dma_mapping);
```

Comment:
nv-p2p.h is clear about that, nvidia_p2p_free_dma_mapping is not required. see below:

/*                                                                                                                                                                                                                 
 * @brief                                                                                                                                                                                                          
 *   Unmap the physical pages previously mapped to the third-party device by                                                                                                                                       
 *   nvidia_p2p_dma_map_pages().                                                                                                                                                                                   
 *                                                                                                                                                                                                                 
 ...                                                                                                                                                                                     
 *   The DMA mapping containing the DMA addresses used by the third-party                                                                                                                                          
 *   device, as retrieved with nvidia_p2p_dma_map_pages(). After this call                                                                                                                                         
 *   returns, neither this struct nor the addresses contained within will be                                                                                                                                       
 *   valid for use by the third-party device.                                                                                                                                                                      
 *                                                                                                                                                                                                                 
 * @return                                                                                                                                                                                                         
 *    0           upon successful completion.                                                                                                                                                                      
 *    -EINVAL     if an invalid argument was supplied.                                                                                                                                                             
 *    -EIO        if an unknown error occurred.                                                                                                                                                                    
 */
int nvidia_p2p_dma_unmap_pages(struct pci_dev *peer,
        struct nvidia_p2p_page_table *page_table,
        struct nvidia_p2p_dma_mapping *dma_mapping);